DEVDOCS-3621: [revise] Shipping Provider API, split doc in two (#1532)
bigcommerce/dev-docs: docs/api-docs/store-management/shipping/use-real-time-carrier.md

# Use a Real-Time Carrier

> Real-time carriers allow merchants and shoppers to obtain **real-time** shipping quotes. There are two types of real-time carriers: native integrations and shipping provider apps. Shipping provider apps require merchants to install the app on their store. Once a merchant installs a shipping provider app on their store, API users and merchants can connect the carrier to the store. API users and merchants can then define and enable a shipping method for a carrier in one or more shipping zones. They can define and enable a different shipping method in each shipping zone for the same carrier. Enabling a zone's real-time shipping method makes the method available to shoppers in that zone during checkout. 

Reviewer: we don't exactly write novels, but it's still key to grab and orient the reader in the first paragraph to help them mentally organize the information they're about to vaguely scan (until they run into issues and pore over it).

revise your opening sentences/paragraphs. Less does more.

Author: added a figure to replace the paragraph

> This article is a guide for API users on how to manage carrier connections and real-time shipping methods from carriers using the [Shipping V2 API](/api-reference/store-management/shipping-api).

Reviewer: what do you intend `API users` to mean?

Author: it's people who use the API, so i changed the wording to reflect this

## Prerequisites
- For shipping provider apps only: Merchant has installed the shipping provider app. 

> - You have obtained the required [OAuth Scopes](/api-docs/getting-started/authentication/rest-api-authentication#oauth-scopes): `Information & Settings`.

Reviewer: you don't obtain OAuth scopes; you create or use API accounts configured with those scopes. Review the new auth docs (the two in the `getting started` section of the /docs/ toc) to get more familiar with the concept and language.

think about who these prerequisites are for. you might want to divide this into two sections, one for provider app developers and one for merchants. I'm pretty sure provder app devs will use app API accounts and merchants will use store API accounts. They also have different questions and concerns as they read.

Author: The doc is for people who are using the REST management APIs, so i made the prerequisites reflect this (store API account).

- You or the merchant have created shipping zones. You can create zones with the [Create a shipping zone](/api-reference/store-management/shipping-api/shipping-zones/createashippingzones) endpoint.

<!-- theme:info -->

> > #### Note

Reviewer: ```suggestion
> #### Native integrations
```

> > USPS, FedEx, and UPS carriers are native-integrations. For a full list of native integrations, see the [Real-Time Shipping Providers](https://support.bigcommerce.com/s/article/Setting-Up-a-Real-Time-Shipping-Quote-Shipping-Method#providers) article.

Reviewer: ```suggestion
> BigCommerce offers native integrations with several shipping services, including FedEx, UPS, and select national postal services. For a full list of native integrations, consult our Help Center article on [Setting up Real Time Shipping Quotes](https://support.bigcommerce.com/s/article/Setting-Up-a-Real-Time-Shipping-Quote-Shipping-Method#providers).
```

## Manage connections to shipping carriers 

> When a merchant installs a shipping provider app, the carrier appears in the merchant's control panel under real-time shipping methods, but is not connected to the store by default. You must first connect the carrier to the store.

Reviewer: ```suggestion
When a merchant installs a shipping provider app, the carrier appears in the <<first menu item they click bold/capitalize>> section of the store control panel, in the <<look up the exact name of the menu item and bold/capitalize>> view. However, the merchant must configure a connection with the carrier before they or their shoppers can request real-time quotes.
```

![Connected vs Installed Carrier](https://storage.googleapis.com/bigcommerce-production-dev-center/images/Installed%20vs%20Connected%20Carrier.png 'Connected versus Installed Carrier. USPS is connected, while FedEx is only installed.')  

### Create a connection

> To set up a carrier using the API, connect it using the [Create a Carrier Connection](/api-reference/store-management/shipping-api/shipping-carrier/postshippingcarrierconnection) endpoint. Make a request containing the ID of the carrier and any configuration fields that the carrier requires. All configuration fields are unique to each carrier. 

Reviewer: ```suggestion
To use the API to connect with a carrier, use the [Create a carrier connection](/api-reference/store-management/shipping-api/shipping-carrier/postshippingcarrierconnection) endpoint. Make a request containing the ID of the carrier and any configuration fields that the carrier requires. Configuration fields are unique to each carrier. 
```

<!--
type: tab
title: Request
-->

> ```json title="Example POST request with X-Auth-Token header" lineNumbers

Reviewer: no need to specify the authentication header in the title

POST https://api.bigcommerce.com/stores/{{STORE_HASH}}/v2/shipping/carrier/connection
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{	
  "carrier_id" : "endicia",
  "connection": {
      "account_id" : "example_id",
      "pass_phrase" : "example_passphrase"
  }
}
```

<!--
type: tab
title: Response
-->

```json title="Example POST response" lineNumbers

> No content

Reviewer: give the HTTP status code when a successful request doesn't contain a body.

Author: added 204 HTTP status code

```

<!-- type: tab-end -->

If a carrier doesn’t require any connection settings, send an empty object for the `connection` property.

<!--
type: tab
title: Request
-->

> ```json title="Example POST request with X-Auth-Token header" lineNumbers

Reviewer: see previous comment

Author: changed the code block titles. i kept it as json so the body would be highlighted correctly

POST https://api.bigcommerce.com/stores/{{STORE_HASH}}/v2/shipping/carrier/connection
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{
  "carrier_id": "example_carrier",
  "connection": {}
}
```

<!--
type: tab
title: Response
-->

```json title="Example POST response" lineNumbers
No content
```

<!-- type: tab-end -->

### Update a connection

> When you update a connection, use the same fields for a carrier as [Create a connection](#create-a-connection). 

Reviewer: ```suggestion
When you update a connection, use the same configuration fields as [Create a connection](#create-a-connection). 
```
make a choice about calling them `connection` fields vs. `configuration` fields, and stick with it.

<!--
type: tab
title: Request
-->

> ```json title="Example PUT request with X-Auth-Token header" lineNumbers

Reviewer: see previous; that goes for all code block headers.

Author: changed the code block titles. i kept it as json so the body would be highlighted correctly

PUT https://api.bigcommerce.com/stores/{{STORE_HASH}}/v2/shipping/carrier/connection
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{	
  "carrier_id" : "endicia",
  "connection": {
      "account_id" : "example_id_2",
      "pass_phrase" : "example_passphrase"
  }
}
```

<!--
type: tab
title: Response
-->

```json title="Example PUT response" lineNumbers
No content
```

<!-- type: tab-end -->

### Delete a connection

> To delete a connection, specify `carrier_id` in the request body.

Reviewer: ```suggestion
To delete a connection, send the `carrier_id` in the request body.
```

<!--
type: tab
title: Request
-->

```json title="Example DELETE request with X-Auth-Token header" lineNumbers
DELETE https://api.bigcommerce.com/stores/{{STORE_HASH}}/v2/shipping/carrier/connection
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{

>   "carrier_id": "example_carrier"

Reviewer: ```suggestion
  "carrier_id": "<<whatever-kind-of-string-a-carrier-id-typically-is>>"
```
consider this.

Author: The carrier_id is usually the name of the carrier (fedex, endicia, etc)

}
```

<!--
type: tab
title: Response
-->

```json title="Example DELETE response" lineNumbers

> No content

Reviewer: http code as previous

Author: added http code

```

<!-- type: tab-end -->

## Manage shipping methods

> After you connect a carrier, you must define and enable a shipping method for the carrier in one or more shipping zones. You can define and enable a different shipping method in each shipping zone for the same carrier.    

Reviewer: do less.

```suggestion
After you connect a carrier, set up a shipping method that uses the carrier in one or more shipping zones. You can define and enable a different shipping method for each zone the carrier services.    
```
I'd think some carriers don't ship to all possible zones.

      "carrier_options": {
          "show_transit_time": "1",
          "packaging_type": "FlatRateLegalEnvelope",
          "delivery_services": [
              "PriorityExpress",
              "ParcelSelect",
              "MediaMail"
          ]
      }
  },
  "enabled": true,
  "handling_fees": {
      "fixed_surcharge": "0"
  }
}
```

<!--
type: tab
title: Response
-->

```json title="Example POST response" lineNumbers
{
   "id": "29",
   "name": "USPS",
   "type": "endicia",
   "settings": {
      "carrier_options": {
         "show_transit_time": "1",
         "packaging": "FlatRateLegalEnvelope",
         "packaging_type": "FlatRateLegalEnvelope",
         "delivery_services": [
            "PriorityExpress",
            "ParcelSelect",
            "MediaMail"
         ]
      }
   },
   "enabled": "true",
   "handling_fees": {
      "fixed_surcharge": "0"
   },
   "is_fallback": "false"
}
```

<!-- type: tab-end -->

For more carrier options, see the [Create a shipping method](/api-reference/store-management/shipping-api/shipping-method/createashippingmethod) reference. Contact the carrier to get a full list of available carrier options.

<!-- theme:info -->

> > After you enable a connected carrier, you can obtain its real-time shipping quotes by using the [Request shipping rates](/api-reference/providers/shipping-provider-api/shipping-provider/requestshippingrates) endpoint. 

Reviewer: does this need to be a callout?

          "packaging_type": "FlatRateLegalEnvelope",
          "delivery_services": {
              "value": "MediaMail"
          }
      }
   },
   "enabled": "true",
   "handling_fees": {
      "fixed_surcharge": "0"
   },
   "is_fallback": "false"
}
```

<!-- type: tab-end -->

### Get a shipping method

<!--
type: tab
title: Request
-->

```json title="Example GET request with X-Auth-Token header" lineNumbers
GET https://api.bigcommerce.com/stores/{{STORE_HASH}}/v2/shipping/zones/{zone_id}/methods/{method_id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json
```

<!--
type: tab
title: Response
-->

```json title="Example GET response" lineNumbers
{
  "id": "29",
  "name": "USPS",
  "type": "endicia",
  "settings": {
      "carrier_options": {
          "show_transit_time": "1",
          "packaging": "FlatRateLegalEnvelope",
          "packaging_type": "FlatRateLegalEnvelope",
          "delivery_services": [
              "MediaMail"
          ]
      }
   },
   "enabled": "true",
   "handling_fees": {
      "fixed_surcharge": "0"
   },
   "is_fallback": "false"
}
```

<!-- type: tab-end -->

<!-- theme:info -->
> #### Note

> > It is also possible to [Get all shipping methods](/api-reference/store-management/shipping-api/shipping-method/getshippingmethodszone).

Reviewer: again, I don't think this needs to be a callout.

```suggestion
You can also [Get all shipping methods](/api-reference/store-management/shipping-api/shipping-method/getshippingmethodszone) in one request.
```
